provider/spark.go

```diff
@@ -829,6 +836,31 @@ func updateQuery(query string, mapping []SourceMapping) (string, []string, error
 	return updatedQuery, sources, nil
 }
 
+func (spark *SparkOfflineStore) getSourcePath(path string) (string, error) {
+	filePaths := strings.Split(path[len("s3://"):], "/")
+	fileType, fileName, fileVariant := filePaths[2], filePaths[3], filePaths[4]
```

Comment:
Is this the format we store this? Or can a user give this path?

Reply:
It is a format we store.